equinoxfitness/datacoco-cloud
# Feature/datcore 888/s3 to s3 utility

## Description

Added a utility to do S3 to S3 Interactions having functions to **copy** and **move** objects from one S3 account to another.
[Ticket Link](https://equinoxfitness.atlassian.net/browse/DATCORE-888)

## Changes

- Added new class S3toS3Interaction
- Added test file for S3toS3Interaction class

## Dependencies
 **None**

### How Has This Been Tested

1. Verify Unit Test Run
- [x]  Switch to virtual environment 
- [x]  Install `requirements.txt`
- [x]  Run `python -m unittest tests/unit/test_s3_to_s3_interaction.py`
- [ ]  Run `python -m unittest discover tests`

2. Verify that the aws service works
- [ ]  Create a python file from the root directory and add  `from datacoco_cloud.s3_to_s3_interaction import S3toS3Interaction`
- [ ]  Initialize the `S3toS3Interaction` class with the source and target access key/secret pair
- [ ]  Call the `duplicate_objects` method
- [ ]  Check the files in the target bucket and folder if files are copied there
- [ ]  Call the `move_objects` method
- [ ]  Check the files in the target bucket and folder if files are copied there and check the files from the source bucket folder if the files are removed

3. Run `tox`
- [ ]   Verify that the test runs successfully

4. Run `pylint datacoco_cloud`
- [ ]   Verify that pylint runs without issue

### Rollback Plan
Check the logs for the files affected.
Check the S3 Directories used in the testing. 

**Source Bucket**
- restore the files missing from the directory that is printed on the logs

**Target Bucket**
- remove the files not originally there from the directory
